src/linker/Linker/UnconditionalSuppressMessageAttributeState.cs

```diff
@@ -99,10 +107,116 @@ private static bool TryDecodeSuppressMessageAttributeData (CustomAttribute attri
 			info = default;
-			return _localSuppressions.TryGetValue (provider, out suppressions) &&
+			if (provider == null)
+				return false;
+
+			return _localSuppressions.TryGetValue (provider, out var suppressions) &&
 				suppressions.TryGetValue (id, out info);
 		}
+
+		private bool IsGloballySuppressed (int id, ICustomAttributeProvider provider, out SuppressMessageInfo info)
+		{
+			DecodeGlobalSuppressMessageAttributes ();
+			return _globalSuppressions.HasGlobalSuppressions (id, provider, out info);
+		}
+
+		private void DecodeGlobalSuppressMessageAttributes ()
+		{
+			if (_globalSuppressions == null) {
+				var suppressions = new GlobalSuppressions ();
+				HashSet<ModuleDefinition> modules = new HashSet<ModuleDefinition> ();
+				foreach (var assembly in _context.Resolver.AssemblyCache.Values) {
+					DecodeGlobalSuppressMessageAttributes (assembly, suppressions);
+					modules.Add (assembly.MainModule);
+				}
+
+				foreach (var module in modules)
+					DecodeGlobalSuppressMessageAttributes (module, suppressions);
+
+				_globalSuppressions = suppressions;
+			}
+		}
+
+		private void DecodeGlobalSuppressMessageAttributes (ICustomAttributeProvider provider, GlobalSuppressions globalSuppressions)
+		{
+			var attributes = provider.CustomAttributes.Where (a => a.AttributeType.Name == "UnconditionalSuppressMessageAttribute");
+			foreach (var instance in attributes) {
+				SuppressMessageInfo info;
+				if (!TryDecodeSuppressMessageAttributeData (instance, out info))
+					continue;
+
+				ModuleDefinition module = null;
+				if (provider is ModuleDefinition _module)
+					module = _module;
+				else if (provider is AssemblyDefinition assembly)
+					module = assembly.MainModule;
+				else if (provider is TypeDefinition type)
+					module = type.Module;
+				else if (provider is IMemberDefinition member)
+					module = member.DeclaringType.Module;
+				else {
+					_context.LogMessage (MessageContainer.CreateInfoMessage ("`UnconditionalSuppressMessage` attribute was placed in an language element which is currently not supported."));
+				}
+
+				Debug.Assert (module != null);
+				if (info.Target == null && (info.Scope == "module" || info.Scope == null)) {
+					AddLocalSuppression (instance, module);
+					continue;
+				}
+
+				if (info.Target == null || info.Scope == null)
+					continue;
+
+				switch (info.Scope) {
+				case "type":
+				case "member":
+					foreach (var result in DocumentationSignatureParser.GetMembersForDocumentationSignature (info.Target, module))
+						globalSuppressions.AddGlobalSuppression (result, info);
+
+					break;
+				case "resource":
+				case "module":
+				case "namespace":
+				case "namespaceanddescendants":
+					_context.LogMessage (MessageContainer.CreateInfoMessage ($"Scope `{info.Scope}` used in `UnconditionalSuppressMessage` is currently not supported."));
+					break;
+				default:
+					break;
+				}
+			}
+		}
+
+		private class GlobalSuppressions
```

Comment:
Please fold that into parent class

Reply:
I think I don't understand what you mean by folding this into the parent class. Could you please clarify?

Reply:
This is a class with like 10 lines where most global logic is in parent class already. You could make for change `IsGloballySuppressed` to have full implementation and it'd make the code more readable.